### PR TITLE
`SharedDirectoryWriteResponse`

### PR DESCRIPTION
**Pull request by ibeckermayer**
Requires backport to v9/v10

#### `webapps`
1. Change the value below to `true` https://github.com/gravitational/webapps/blob/becfad38776a68e0d8be634476f24cead7890ffc/packages/teleport/src/config.ts#L32
2. run the development server like `yarn start-teleport --target=https://ec2-35-171-27-185.compute-1.amazonaws.com/`
3. See https://gravitational.slack.com/archives/C02DQ1C2BMW/p1657807702085829 for a username/password combo

## What to expect

Go to the top right `...` menu and click `Share Directory`

image

(X-out of the "uncaught exception" screen if you get it). The directory you select should show up in the File Explorer as a shared drive

Screen Shot 2022-07-19 at 14 13 45

The drive should be navigable as well should any folders within it

image

Trying to open a file should show you the file contents (if you open it in Notepad for example). You will be able to type into the file, and if you save the file the changes should show up on your local disk.

### CODE REVIEW

**Review by hatched — APPROVED** [review]

**Comment by kimlisa** [comment]
was that `document is not focused` error happening during directory sharing? or just in general?

**Comment by ibeckermayer** [comment]
> resizing isn't possible yet right?

correct

> was that `document is not focused` error happening during directory sharing? b/c i get that error even when i'm not sharing (but i don't remember the original discussion).

Yes that error was only ever happening for me right around when I started directory sharing. You're seeing it while just clicking around the interface while not directory sharing?

@lisa

**Comment by lisa** [comment]
> > resizing isn't possible yet right?
> 
> correct
> 
> > was that `document is not focused` error happening during directory sharing? b/c i get that error even when i'm not sharing (but i don't remember the original discussion).
> 
> Yes that error was only ever happening for me right around when I started directory sharing. You're seeing it while just clicking around the interface while not directory sharing?
> 
> @lisa

# `@lisa` User Notification

Hello! You're receiving this notice because you've tagged me, @lisa, on GitHub, likely in error. Lisa is a common first name, and happens to be mine. I use the @lisa username here on GitHub. While being tagged in various repositories is an interesting way for me to discover new projects, it's likely not your intention. Did you mean to tag someone else? Did you mean to use `@lisa` instead, in a [word as word](https://en.wikipedia.org/wiki/Use%E2%80%93mention_distinction) usage?

# Next Steps

I'm unsubscribing from this issue, pull request or project so if you meant to tag someone else to notify them, I would recommend that. If you meant to use my user ID as a plain string, without having GitHub notify me, you can do so by wrapping with backticks (the \` character), like so: `@lisa`.

File: packages/teleport/src/lib/tdp/client.ts [hunk 1]
```diff
@@ -274,12 +275,20 @@ export default class Client extends EventEmitterWebAuthnSender {
...
 
-  handleSharedDirectoryWriteRequest(buffer: ArrayBuffer) {
+  async handleSharedDirectoryWriteRequest(buffer: ArrayBuffer) {
     const req = this.codec.decodeSharedDirectoryWriteRequest(buffer);
-    // TODO(isaiah): delete debug logs
-    this.logger.debug('Received SharedDirectoryWriteRequest:');
-    this.logger.debug(req);
-    // TODO(isaiah): here's where we'll respond with a SharedDirectoryWriteResponse
+
+    const bytesWritten = await this.sdManager.writeFile(
```
[the remark this answers is not in the document]
Comment (ibeckermayer) [reply packages/teleport/src/lib/tdp/client.ts:281]: Yes good catch. Typically anything that calls `this.sdManager` should be wrapped in `try/catch`

https://github.com/gravitational/webapps/blob/11eb8ed5d3b88b8a1433e629cb931b56c4a53da6/packages/teleport/src/lib/tdp/sharedDirectoryManager.ts#L15-L17

**Pull request merged**